src/connectors/h1_collect.py

```diff
@@ -0,0 +1,316 @@
+    ('state', 'string'),
+    ('created_at', 'TIMESTAMP_LTZ(9)'),
+    ('vulnerability_information', 'string'),
+    ('triaged_at', 'TIMESTAMP_LTZ(9)'),
+    ('closed_at', 'TIMESTAMP_LTZ(9)'),
+    ('first_program_activity_at', 'TIMESTAMP_LTZ(9)'),
+    ('bounty_awarded_at', 'TIMESTAMP_LTZ(9)'),
+    ('rating', 'string'),
+    ('name', 'string'),
+    ('description', 'string'),
+    ('external_id', 'string'),
+    ('asset_identifier', 'string'),
+    ('awarded_amount', 'int'),
+    ('awarded_bonus_amount', 'int'),
+]
+
+
+# Perform the API call
+def load_data(url: str, token: str, api_identifier: str, params: dict = {}) -> dict:
+    headers: dict = {"Accept": "application/json"}
+
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        resp = requests.get(
+            url, auth=(api_identifier, token), params=params, headers=headers
+        )
+        resp.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"Error GET: url={url}")
+        log.error(f"HTTP error occurred: {http_err}")
+        raise
+    log.debug(resp.status_code)
+    return resp.json()['data']
+
+
+def get_path(data, path, default=None):
+    for key in path.split("."):
+        if key not in data:
+            return default
+        else:
+            data = data[key]
+    return data
+
+
+def connect(connection_name, options):
+    landing_transactions_table = f'data.h1_collect_transactions_connection'
+    landing_reports_table = f'data.h1_collect_reports_connection'
+
+    comment = yaml_dump(module='h1_collect', **options)
+
+    db.create_table(
+        name=landing_transactions_table,
+        cols=LANDING_TABLE_COLUMNS_TRANSACTIONS,
+        comment=comment,
+    )
+    db.execute(
+        f'GRANT INSERT, SELECT ON {landing_transactions_table} TO ROLE {SA_ROLE}'
+    )
+
+    db.create_table(
+        name=landing_reports_table,
+        cols=LANDING_TABLE_COLUMNS_REPORTS,
+        comment=comment,
+    )
+    db.execute(f'GRANT INSERT, SELECT ON {landing_reports_table} TO ROLE {SA_ROLE}')
+    return {
+        'newStage': 'finalized',
+        'newMessage': "HackerOne reports ingestion table created!",
+    }
+
+
+def ingest(table_name, options, dryrun=False):
+    ingest_type = (
+        'transaction' if table_name.endswith('_TRANSACTIONS_CONNECTION') else 'report'
+    )
+    landing_table = f'data.{table_name}'
+
+    timestamp = datetime.utcnow()
+    api_identifier = options['api_identifier']
+    api_token = options['api_token']
+    account_id = options['account_id']
+    program_name = options['program_name']
+
+    if ingest_type == 'transaction' and account_id:
+        try:
+            transactions = load_data(
+                f'https://api.hackerone.com/v1/programs/{account_id}/billing/transactions',
+                api_token,
+                api_identifier,
+                params=None,
+            )
+        except requests.exceptions.HTTPError as e:
+            log.error(e)
```

Comment:
this will log an error to CloudWatch, but not record it in the metadata tables. the runners that calls into this expects exceptions to be raised if there is an error and handle them [here](https://github.com/snowflakedb/SnowAlert/blob/master/src/runners/connectors_runner.py#L107).

in general, try / except should be used in code that decides how to handle errors, and so we want to do it in a general way by the runner framework rather than in each handler.

Reply:
To clarify, we should remove the try/except (lines 151-159) and let the runner handle the errors? Or should we do a raise at line 160 to pass the error to the runners as well? ie where should the error be logged?

Reply:
up to you, but I would just remove. the runners will log the error.